src/bin/ion/commands/generate/generator.rs

```diff
@@ -336,6 +338,7 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
         isl_type_name: &String,
         isl_type: &IslType,
         code_gen_context: &mut CodeGenContext,
+        is_nested_type: bool,
```

Comment:
Doesn't need to be addressed in this PR, but I think that our data model might need to include a distinction between the "new type" pattern and "primitive type" rather than just having a `is_nested_type` field.

Reply:
This is_nested type is not necessarily distinction between new type and primitive because even a user defined type when used as a type reference will not be nested type but still a new type right?
e.g.
```
type::{
  name: my_type,
  type: foo // here `foo` is a type reference but not necessarily a nested type
}
```

Reply:
> ```
> type::{
>  name: my_type,
>  type: foo // here `foo` is a type reference but not necessarily a nested type
>}
>```
I think this would be a new-type.


Whether or not a type is "nested" is just a logistical thing that relates to whether we have to generate a name for it because it's anonymous. (Since we need to generate a name, we also put it in a namespace to ensure that we don't accidentally run into name conflicts.)

That being said, as our code generator gets more sophisticated, we don't have to generate a wrapper type for things like this:
```
type::{
  name: foo,
  fields: {
    bar: { type: int, valid_values: range::[1, max] }
  }
}
```
Rather than creating a `Bar` type, we could just use `Integer` (or `u64` or whatever), and add the range validation into the constructor and/or builder of `Foo`.

We would still need to create a nested type for cases like this...
```
type::{
  name: foo,
  fields: {
    bar: { 
      fields: {
        baz: int,
        quuz: bool,
      }
    }
  }
}
```
... because even though ISL is strictly subtractive, the type of the `bar` field is a structural composition of other types rather than just being a simple restriction on an existing type.